Data/AuthenticationClient/Sources/Persistance.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  Persistance.swift
+//  QuranEngine
+//
+//  Created by Mohannad Hassan on 28/12/2024.
+//
+
+import Foundation
+import VLogging
+
+enum PersistanceError: Error {
+    case persistanceFailed
+    case retrievalFailed
+}
+
+/// An abstraction for secure persistance of the authentication state.
+protocol Persistance {
```

Comment:
I suggest creating a general `SecurePersistence` API (under Core/ directory) rather than one specifically for authentication.  

A recommended pattern can be seen in [this directory](https://github.com/quran/quran-ios/tree/3d34dadc1fb260e035b23649fd28f5b06a22f6b0/Core/Preferences), where `Preferences` is implemented as a service, and consumers define `PreferenceKey` objects to describe the data that needs to be persisted. I think we can do something similar with `SecurePersistence` and `SecurePersistenceKey`.

Also please add a couple of tests for this class.